### PR TITLE
instancetype: Require Guest within MemoryInstancetype

### LINKED ISSUE #8171: instancetype: MemoryInstancetype.Guest not required
**What happened**:

`Guest` within `MemoryInstancetype` is not a required field while `Memory` is within `VirtualMachineInstancetypeSpec`.

**What you expected to happen**:

`Guest` should be requried within `MemoryInstancetype`.

**How to reproduce it (as minimally and precisely as possible)**:
```
# cat examples/csmall.yaml
---
apiVersion: instancetype.kubevirt.io/v1alpha1
kind: VirtualMachineInstancetype
metadata:
  name: csmall
spec:
  cpu:
    guest: 1 
  memory: {}
# ./cluster-up/kubectl.sh apply -f examples/csmall.yaml
# ./cluster-up/kubectl.sh apply -f examples/vm-cirros-csmall.yaml
# ./cluster-up/virtctl.sh start vm-cirros-csmall
# ./cluster-up/kubectl.sh get vms/vm-cirros-csmall -o json | jq .status.conditions[1].message
selecting docker as container runtime
"Failure while starting VMI: admission webhook \"virtualmachineinstances-create-validator.kubevirt.io\" denied the request: no memory requested, at least one of 'spec.domain.memory.guest', 'spec.domain.memory.hugepages.size' or 'spec.domain.resources.requests.memory' must be set"
```

**Environment**:
- KubeVirt version (use `virtctl version`): main
- Kubernetes version (use `kubectl version`): N/A
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by lyarwood**
CPUInstancetype and MemoryInstancetype are both required attributes of
VirtualMachineInstancetypeSpec however only CPUInstancetype would
previously require that an amount of guest visible resources were
provided through its' Guest attribute.

This change corrects this for MemoryInstancetype ensuring that an amount
of guest visible memory is always provided with an instancetype as per
the original design.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8171

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/kubevirt.io/api/instancetype/v1alpha1/types.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ type CPUInstancetype struct {
 type MemoryInstancetype struct {
 
 	// Required amount of memory which is visible inside the guest OS.
-	Guest *resource.Quantity `json:"guest,omitempty"`
+	Guest *resource.Quantity `json:"guest"`
```
Comment (akrejcir) [inline staging/src/kubevirt.io/api/instancetype/v1alpha1/types.go:164]: This field doesn't need to be a pointer, because it is required.

Comment (lyarwood) [reply staging/src/kubevirt.io/api/instancetype/v1alpha1/types.go:164]: ACK fair point, I'll rework shortly.

**Pull request merged**